app/components/radio_button/radio_button_group.js

```diff
@@ -22,29 +22,26 @@ export default class RadioButtonGroup extends PureComponent {
     constructor(props) {
         super(props);
 
-        this.selected = null;
-        if (this.props.options.length) {
-            this.props.options.forEach((option) => {
-                const {
-                    value,
-                    checked,
-                } = option;
+        this.state = {
+            selected: this.getSelectedValue(),
+        };
+    }
 
-                if (!this.state.selected && checked) {
-                    this.selected = value;
-                }
-            });
+    componentWillReceiveProps(nextProps) {
+        if (this.props.options !== nextProps.options) {
+            this.setState({selected: this.getSelectedValue()});
         }
-
-        this.state = {selected: this.selected};
     }
 
-    get value() {
-        return this.state.selected;
-    }
+    getSelectedValue = (options = []) => {
+        let selected;
+        for (const option in options) {
+            if (option.checked) {
+                selected = option.value;
```

Comment:
can't you break once you find the checked value?

Reply:
Done (added `break`)